### PR TITLE
haskellPackages: update stackage and hackage

### PR DESCRIPTION
**Pull request by maralorn**
### This Merge

This PR is the regular merge of the `haskell-updates` branch into `master`.

This branch is being continually built and tested by hydra at https://hydra.nixos.org/jobset/nixpkgs/haskell-updates. You may be able to find an up-to-date Hydra build report at [cdepillabout/nix-haskell-updates-status](https://github.com/cdepillabout/nix-haskell-updates-status).

We roughly aim to merge these `haskell-updates` PRs at least once every two weeks. See the @NixOS/haskell [team calendar](https://cloud.maralorn.de/apps/calendar/p/H6migHmKX7xHoTFa) for who is currently in charge of this branch.

### haskellPackages Workflow Summary

Our workflow is currently described in [`pkgs/development/haskell-modules/HACKING.md`](https://github.com/NixOS/nixpkgs/blob/haskell-updates/pkgs/development/haskell-modules/HACKING.md).

The short version is this:
* We regularly update the Stackage and Hackage pins on `haskell-updates` (normally at the beginning of a merge window).
* The community fixes builds of Haskell packages on that branch.
* We aim at at least one merge of `haskell-updates` into `master` every two weeks.
* We only do the merge if the [`mergeable`](https://hydra.nixos.org/job/nixpkgs/haskell-updates/mergeable) job is succeeding on hydra.
* If a [`maintained`](https://hydra.nixos.org/job/nixpkgs/haskell-updates/maintained) package is still broken at the time of merge, we will only merge if the maintainer has been pinged 7 days in advance. (If you care about a Haskell package, become a maintainer!)

---

This is the follow-up to #181822. Come to [#haskell:nixos.org](https://matrix.to/#/#haskell:nixos.org) if you have any questions.

### CODE REVIEW

**Comment by roberth** [comment]
https://github.com/cdepillabout/nix-haskell-updates-status reports no maintained breakages. Does that mean that this PR can be merged?

We can upgrade the hercules-ci / cachix `nix` pins in the next iteration.

**Comment by cdepillabout** [comment]
@roberth Yes, since the build report is passing, we can merge this in.

Normally we just wait for the current person watching `haskell-updates` to merge it in (which is @maralorn), but sometimes other people do the merge if they particularly want something (and they first clear it with the maintainer on duty).

Would you be interested in trying to merge in? I just talked to @maralorn, and he is fine if you want to take a shot.

The general steps are specified in `HACKING.md`: https://github.com/NixOS/nixpkgs/blob/81fedf1c285c231bf2958dc4b3ecb00af2b5d01a/pkgs/development/haskell-modules/HACKING.md

In particular, https://github.com/NixOS/nixpkgs/blob/81fedf1c285c231bf2958dc4b3ecb00af2b5d01a/pkgs/development/haskell-modules/HACKING.md#script-for-merging-haskell-updates-and-opening-a-new-pr.

The tl;dr steps are:

1. Make sure there are no eval errors on the `haskell-updates` jobset on hydra.
2. `gh auth login`
3. Get hydra repor (`./maintainers/scripts/haskell/hydra-report.hs get-report`), mark newly broken Haskell packages (`./maintainers/scripts/haskell/mark-broken.sh --do-commit`), and push that commit.
4. Run a hydra eval to double check that no new eval errors have been generated.
5. Run `./maintainers/scripts/haskell/merge-and-open-pr.sh 182019` to merge this PR and open the next one.

(If you want to try a merge, you might want to ping @maralorn real quick either on here or on matrix just to make sure he's not attempting a merge at the same time.)

**Comment by roberth** [comment]
Thanks @cdepillabout for the explanation. I didn't get around to it that quickly, so thanks @maralorn for the merge and next PR :heart:
